hass-dash.el

```diff
@@ -244,29 +259,42 @@ the font face for the title."
 
 ;; User functions
 ;;;###autoload
-(defun hass-dash-refresh ()
-  "Rerender the hass-dash buffer."
-  (interactive)
-  (with-current-buffer (get-buffer-create hass-dash-buffer-name)
+(defun hass-dash-refresh (dashboard)
+  "Rerender the hass-dash buffer for DASHBOARD."
+  (with-current-buffer (get-buffer-create (hass-dash-key-to-buffer-name dashboard))
     (let ((inhibit-read-only t)
           (prev-line (line-number-at-pos)))
       (erase-buffer)
-      (hass-dash--track-layout-entities (widget-create (append '(group :format "%v")
-                                                               hass-dash-layout)))
+      (hass-dash--track-layout-entities
+       (widget-create
+        (append '(group :format "%v")
+                (cdr (assoc dashboard hass-dash-layouts)))))
       (goto-char (point-min))
       (forward-line (1- prev-line))
       (hass-dash-mode))))
 
-;;;###autoload
-(defun hass-dash-open ()
-  "Open the hass-dash buffer."
+(defun hass-dash-refresh-current ()
+  "Rerender the current hass-dash buffer."
   (interactive)
-  (hass-dash-refresh)
-  (let* ((buffer (get-buffer hass-dash-buffer-name))
+  (let ((key (hass-dash-buffer-name-to-key (buffer-name))))
+    (when key
+      (hass-dash-refresh key))))
+
+;;;###autoload
+(defun hass-dash-open (dashboard)
+  "Open the hass-dash buffer for DASHBOARD."
+  (hass-dash-refresh dashboard)
+  (let* ((buffer (get-buffer (hass-dash-key-to-buffer-name dashboard)))
          (window (get-buffer-window buffer)))
```

Comment:
I feel like the different layout should still use a single buffer, or at least re-use the window if it's open. I don't see myself opening multiple dashboards at a time. I'm open about this and curious on your use.

Reply:
So, I don't see myself using it either.  However, I still opted for this approach because I could see a use case for it, and I don't really see many downsides (buffers are cheap after all).

If we wanted to make this easy to be user-configurable, we can always store an alist mapping buffer names to keys so we don't have to parse the key out of the buffer name in `hass-dash-buffer-name-to-key`.  Then, you'd be able to make the dashboards use a single buffer by just overriding `hass-dash-key-to-buffer-name`.  Thoughts?